assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.js

```diff
@@ -0,0 +1,218 @@
+import {
+	AdminMenuTooltip,
+	useShowTooltip,
+	useTooltipState,
+} from '../../../../components/AdminMenuTooltip';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import { MODULES_ADSENSE } from '../../datastore/constants';
+import { getTimeInSeconds } from '../../../../util';
+import { ACCOUNT_STATUS_READY, SITE_STATUS_READY } from '../../util';
+import useViewOnly from '../../../../hooks/useViewOnly';
+import {
+	BREAKPOINT_SMALL,
+	useBreakpoint,
+} from '../../../../hooks/useBreakpoint';
+const { useSelect, useDispatch } = Data;
+
+export default function AdBlockingRecoveryWidget( { Widget, WidgetNull } ) {
+	const AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG =
+		'ad-blocking-recovery-notification';
+	const breakpoint = useBreakpoint();
+	const viewOnlyDashboard = useViewOnly();
+
+	const showTooltip = useShowTooltip(
+		AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG
+	);
+	const { isTooltipVisible } = useTooltipState(
+		AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG
+	);
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG
+		)
+	);
+	const adSenseModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'adsense' )
+	);
+	const adBlockingRecoverySetupStatus = useSelect( ( select ) => {
+		if ( viewOnlyDashboard ) {
+			return null;
+		}
+		return select( MODULES_ADSENSE ).getAdBlockingRecoverySetupStatus();
+	} );
+	const accountStatus = useSelect( ( select ) => {
+		if ( viewOnlyDashboard ) {
+			return null;
+		}
+		return select( MODULES_ADSENSE ).getAccountStatus();
+	} );
+	const setupCompletedTimestamp = useSelect( ( select ) => {
+		if ( viewOnlyDashboard ) {
+			return null;
+		}
+		return select( MODULES_ADSENSE ).getSetupCompletedTimestamp();
+	} );
+	const siteStatus = useSelect( ( select ) => {
+		if ( viewOnlyDashboard ) {
+			return null;
+		}
+		return select( MODULES_ADSENSE ).getSiteStatus();
+	} );
+	const learnMoreURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getGoogleSupportURL( {
+			path: '/adsense/answer/11576589',
+		} )
+	);
+	const recoveryPageURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-ad-blocking-recovery' )
+	);
+
+	const { dismissItem } = useDispatch( CORE_USER );
+	const dismissCallback = async () => {
+		await dismissItem( AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG );
+	};
+
+	if ( isTooltipVisible ) {
+		return (
+			<Fragment>
+				<WidgetNull />
+				<AdminMenuTooltip
+					title={ __(
+						'You can always set up ad blocking recovery from Settings later',
+						'google-site-kit'
+					) }
+					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
+					onDismiss={ dismissCallback }
+					tooltipStateKey={ AD_BLOCKING_RECOVERY_NOTIFICATION_SLUG }
+				/>
+			</Fragment>
+		);
+	}
+
+	const THREE_WEEKS_IN_SECONDS = getTimeInSeconds( 'week' ) * 3;
+	const NOW_IN_SECONDS = Math.floor( Date.now() / 1000 );
+	const shouldShowWidget =
```

Comment:
Sorry if I didn't spot this previously, but we can use the same `WEEK_IN_SECONDS` constant to calculate this value (in fact there's not so much reason to keep it at all, we can just inline `WEEK_IN_SECONDS * 3` below).

Also, this mix of naming styles for constants in the middle of the component doesn't look too good, please can you rename using camel-case?

Lastly a super trivial one but an empty line before the `shouldShowWidget` declaration would help the readability.

```suggestion
	const nowInSeconds = Math.floor( Date.now() / 1000 );

	const shouldShowWidget =
```

Reply:
Updated.